### ISSUE TITLE
[Neoforge] Spear with Better Combat acts as a two-handed weapon despite having a tiny model

### ISSUE THREAD

**Issue by v3xed-v0id** [SilentChaos512/Silent-Gear#709]
## Versions

- **Silent Gear**: 4.0.4
- **Silent's Gems**: N/A
- **Silent Lib**: 10.3.0
- **Forge**: Neoforge 21.1.68
- **Modpack**: N/A
- **Optifine Installed**: No

## Expected Behavior

i would expect the spear to act as a one-handed spear and not a two-handed spear, also with it's current stats it doesnt make sense for the spear to be giving up an entire extra weapon slot (bc its two-handed so ur not able to use a shield with it or dual wield) as its stats are more on par with a broadsword than really anything two-handed, i would recommend changing the spear animation to a one-handed spear animation personally, maybe using the trident animation? this seems like the easiest solution personally

alternatively, if you wanted to lean the spear into being two-handed, maybe elongating the model a bit to make it fit two-handed while maybe buffing the weapon reach and lowering swing speed sounds like a good idea (since u gotta keep in mind that spear means ur giving up a weapon/shield slot cus its two-handed, so making it act more like a two-handed weapon in terms of pure stats seems like a good idea, my opinion)

honestly i would be ok with either of these solutions, u pick which one u like more i think, tho the first one is probably easiest to implement cus i doubt you would have to change textures around n stuff and it would still fit lol

(Description here)

## Actual Behavior
Spear instead has a very tiny model, it looks off with better combat when being used since it was assigned to be two-handed while not being given the stats for a proper two-handed weapon as two-handed means you will lack a shield/dual-wield, and the spear was given stats more similar to a one-handed broadsword.

(Description here)

## Links/Images

the problem:
![2024-10-11_00 11 15](https://github.com/user-attachments/assets/e7cbe8a0-6439-40ab-b785-8e11199a8a13)
![2024-10-11_00 11 32](https://github.com/user-attachments/assets/f5362e0a-2030-4cce-9b89-8e9a02d986c3)

solution proposal:
![2024-10-11_00 11 24](https://github.com/user-attachments/assets/4f5a4452-acda-45e6-b4ac-620186140386)
![2024-10-11_00 11 18](https://github.com/user-attachments/assets/27753492-8a77-4e3e-958e-3da74ae45014)

- (Link)

## Steps to Reproduce the Problem

1. boot up game
2. swing spear once acquired
3. observe

**Comment by v3xed-v0id** [comment]
i appreciate you adding better combat support in the latest update btw! :D

**Comment by ffuentesm** [comment]
I think the problem is the animation template used for the spear in this case. 
https://github.com/ZsoltMolnarrr/BetterCombat/blob/1.21.1/common/src/main/resources/data/bettercombat/weapon_attributes/spear.json
You can see that it's marked for two handed. 
`{
  "attributes": {
    "attack_range": 3.5,
    "pose": "bettercombat:pose_two_handed_polearm",
    "two_handed": true,
    "category": "spear",
    "attacks": [
      {
        "hitbox": "FORWARD_BOX",
        "damage_multiplier": 1,
        "angle": 0,
        "upswing": 0.5,
        "animation": "bettercombat:two_handed_stab_right",
        "swing_sound": {
          "id": "bettercombat:spear_stab"
        }
      }
    ]
  }
}`

A way to fix that is to change the "two" for "one" so that it uses the one handed animations. 
As you can see in the trident file, it uses one handed stab:
`{
  "parent": "bettercombat:spear",
  "attributes" : {
    "attack_range": 2.75,
    "pose": "",
    "two_handed": false,
    "category": "trident",
    "attacks": [
      {
        "animation": "bettercombat:one_handed_stab"
      }
    ]
  }
}`

**Comment by v3xed-v0id** [comment]
> I think the problem is the animation template used for the spear in this case. https://github.com/ZsoltMolnarrr/BetterCombat/blob/1.21.1/common/src/main/resources/data/bettercombat/weapon_attributes/spear.json You can see that it's marked for two handed. `{ "attributes": { "attack_range": 3.5, "pose": "bettercombat:pose_two_handed_polearm", "two_handed": true, "category": "spear", "attacks": [ { "hitbox": "FORWARD_BOX", "damage_multiplier": 1, "angle": 0, "upswing": 0.5, "animation": "bettercombat:two_handed_stab_right", "swing_sound": { "id": "bettercombat:spear_stab" } } ] } }`
> 
> A way to fix that is to change the "two" for "one" so that it uses the one handed animations. As you can see in the trident file, it uses one handed stab: `{ "parent": "bettercombat:spear", "attributes" : { "attack_range": 2.75, "pose": "", "two_handed": false, "category": "trident", "attacks": [ { "animation": "bettercombat:one_handed_stab" } ] } }`

yeah the spear would make more sense as one-handed, i think thats probably the better solution and would make the most sense considering the size of it, one-handed small spears exist in real life so its not like it would be inaccurate or anything, imo it would be the best solution
